src/components/UserSelector.tsx

```diff
@@ -2,15 +1,68 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { User } from '../types/User';
+import cl from 'classnames';
+import { pause } from '../utils/methods';
+import { Post } from '../types/Post';
+
+interface Props {
+  users: User[];
+  onActiveUser: (v: User) => void;
+  activeUser: User | null;
+  onShowSideBar: (v: boolean) => void;
+  onActivePost: (v: Post | null) => void;
+}
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  onActiveUser,
+  activeUser,
+  onShowSideBar,
+  onActivePost,
+}) => {
+  const [showDropDown, setShowDropDown] = useState(false);
+  const divRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    const handleClickedOutside = (e: MouseEvent) => {
+      if (divRef.current && !divRef.current.contains(e.target as Node)) {
+        setShowDropDown(false);
+      }
+    };
+
+    document.addEventListener('click', handleClickedOutside);
+
+    return () => {
+      document.removeEventListener('click', handleClickedOutside);
+    };
+  }, []);
+
+  const handleSelectUser = async (user: User) => {
+    onActiveUser(user);
+    setShowDropDown(false);
+    onShowSideBar(false);
+    onActivePost(null);
+    await pause();
```

Comment:
The `pause` function is used here, but it's unclear why it is necessary. Ensure that this is intentional and that it serves a specific purpose in the user selection logic.